File: School Projects/2017 - Fundementals Of CSC 2 (Java)/Ship/CargoShip.java
package ship;
/*
    @author v.shydlonok

    CargoShip class extends the Ship class.
    Uses capacity int to keep track of the capacity of 
        the cargo ship.
    Defines a getter and setter for the capacity variable.
*/
public class CargoShip extends Ship 
{
    protected int capacity;
    
    public CargoShip()
    {
        super();
        capacity = 0;
    }
    
    public CargoShip(int capacity,String name,int year)
    {
        super(name,year);
        this.capacity = capacity;
    }

    public int getCapacity() {
        return capacity;
    }

    public void setCapacity(int capacity) {
        this.capacity = capacity;
    }
}

File: School Projects/2017 - C++ Programming (C++)/Ship/CargoShip.cpp
#include "CargoShip.h"

CargoShip::CargoShip()
{
	SetCargoCapacity(0);
}

CargoShip::CargoShip(string name, int year, int capacity) : Ship(name, year)
{
	SetCargoCapacity(capacity);
}

int CargoShip::GetCargoCapacity() const
{
	return cargoCapacity;
}

void CargoShip::SetCargoCapacity(int capacity)
{
	this->cargoCapacity = capacity;
}

ostream & operator<<(ostream & strm, const CargoShip & obj)
{
	return strm << "Cargo ship name: " << obj.GetName() << endl
				<< "Cargo ship year: " << obj.GetYear() << endl
				<< "Cargo ship cargo capacity: " << obj.GetCargoCapacity() << endl;
}
